### PR TITLE
Express validation

### LINKED ISSUE #121: Add express validation to routes

### PR DESCRIPTION
**Pull request by SivanC**
Adds extra data validation to all express routes. Closes #121.

### CODE REVIEW

File: back-end/users.js [hunk 1]
```diff
@@ -49,27 +50,38 @@ router.get('/:id', async (req, res) => {
...
+    } else {
+      const data = matchedData(req)
+      // Update user information
+      // since this is just an update and not a create we don't need to guarantee
+      // every field.
+      const user = await User.findById(data.id).exec()
+      if (data.email) {
+        user.email = data.email
+      }
+      if (data.firstName) {
```
Comment (byroncjy) [inline back-end/users.js:72]: need to change firstName to first_name, same for lastName
was going to fix in my other PR but now is good

File: back-end/platforms.js [hunk 2]
```diff
@@ -19,23 +20,38 @@ router.get('/:id/platforms', async (req, res) => {
 })
 
 // Route for updating platform information
-router.put('/:id/platforms', async (req, res) => {
+router.put('/:id/platforms', 
+  param('id').trim().notEmpty().isMongoId(),
```
Comment (byroncjy) [inline back-end/platforms.js:24]: Why trim here but not for the id field in users.js put request (line 53)?

**Review by byroncjy — COMMENTED** [review]
Looks good, just change the name field names
Also pull from master and check how it interacts with tests

File: back-end/platforms.js (hunk 2 above)
Comment (SivanC) [reply back-end/platforms.js]: forgot to remove, fixed

**Pull request merged**